### PR TITLE
#187419198 User should interact with each others via chats publicly 

### PR DESCRIPTION
**Pull request by niyobertin**
**What does this PR do?**

Have chats room to interact  with each other to ask information publicly.
**Description of Task to be completed?**

Have the following end point 
GET: /api/v1/messages
**How should this be manually tested?**
After cloning the repo, cd into it and run _npm run dev_  to start the server.

- using your browser vist http://localhost:PORT/api/v1/messages you will be provided by the link to join the chat room.
- provide your username and text your message the send.

**Any background context you want to provide?**
when a user send a message it will be echoed to all online users

**What are the relevant pivotal tracker stories?**

[Delivers #[187419198](https://www.pivotaltracker.com/n/projects/2701065)]

### CODE REVIEW

File: src/routes/chatRoutes.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+        socket.on('chat message', async(data: any) => {
+            await sendMessages(io, data);
+        });
+        
+    });
+}
+export const joinChatRoomRoutes = Router();
+joinChatRoomRoutes.get('/',joiRoom);
+export default chatRoutes;
```
Comment (teerenzo) [inline src/routes/chatRoutes.ts]: you don't need to have this as route just have config of your socket, if you decide to this without route remove this and have it as config of socket

File: src/utils/server.ts [hunk 2]
```diff
@@ -28,9 +37,13 @@ app.use(
...
+app.use("/api/v1/chats", express.static(path.join(__dirname, '../../public')));
+
 app.use("/", homeRoute);
 app.use("/api/v1", appROutes);
 app.use("/docs", docRouter);
 app.use("/api/v1/roles", RoleRouter);
 
-export default app;
+chatRoutes(io);
+
```
Comment (teerenzo) [inline src/utils/server.ts]: this not route have socket as config and you call it here

**Review by teerenzo — CHANGES_REQUESTED** [review]

**Review by yvanddniyo — APPROVED** [review]
LGTM

**Review by MugemaneBertin2001 — APPROVED** [review]
LGTM

**Review by teerenzo — APPROVED** [review]

**Review by teerenzo — APPROVED** [review]

**Pull request merged**